File: myurlshortner-consumer/myurlshortner-consumer/src/main/scala/com/acme/myurlshortner/consumer/domain/entity/useragent/OperatingSystem.scala
package com.acme.myurlshortner.consumer.domain.entity.useragent

sealed trait OperatingSystem(
  val value: String,
  val userAgentValue: String | Nothing
)

case object Windows   extends OperatingSystem("Windows", "Windows")
case object Macintosh extends OperatingSystem("Macintosh", "Macintosh")
case object Linux     extends OperatingSystem("Linux", "Linux")
case object OtherOS   extends OperatingSystem("OtherOS", null)

File: myurlshortner-consumer/src/main/kotlin/com/acme/myurlshortner/consumer/domain/useragent/OperatingSystem.kt
package com.acme.myurlshortner.consumer.domain.useragent

sealed class OperatingSystem {
    val userAgentValue: String?

    constructor(userAgentValue: String?) {
        this.userAgentValue = userAgentValue
    }

    data object Windows : OperatingSystem("Win")
    data object Macintosh : OperatingSystem("Mac")
    data object Linux : OperatingSystem("Linux")
    data object Other : OperatingSystem(null)
}
